src/main/java/com/dev/cinema/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,29 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String login) throws UsernameNotFoundException {
+        User user = userService.findByLogin(login).orElseThrow(()
+                -> new RuntimeException("Could not find user by login" + login));
```

Comment:
let's throw here "...specific Spring exception when user is not found by username"
https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_08.html